src/NLog.Extensions.Logging/Targets/ExtensionILoggerTarget.cs

```diff
@@ -0,0 +1,173 @@
+                    eventId = new Microsoft.Extensions.Logging.EventId(eventId.Id, eventNameValue);
+            }
+
+            var layoutMessage = RenderLogEvent(Layout, logEvent);
+            IDictionary<string, object> contextProperties = null;
+            if (ContextProperties.Count > 0 || IncludeMdlc || IncludeMdc || IncludeGdc)
+            {
+                contextProperties = GetContextProperties(logEvent);
+                if (contextProperties?.Count == 0)
+                    contextProperties = null;
+            }
+
+            _logger.Log(LookupLogLevel(logEvent.Level), eventId, new LogState(logEvent, layoutMessage, contextProperties), logEvent.Exception, LogStateFormatter);
+        }
+
+        struct LogState : IReadOnlyList<KeyValuePair<string, object>>
+        {
+            public int Count => (LogEvent.HasProperties ? LogEvent.Properties.Count : 0) + (ContextProperties?.Count ?? 0) + 1;
+
+            public KeyValuePair<string, object> this[int index]
+            {
+                get
+                {
+                    if (LogEvent.HasProperties)
+                    {
+                        if (index < LogEvent.Properties.Count)
+                        {
+                            foreach (var prop in LogEvent.Properties)
+                            {
+                                if (index-- == 0)
+                                    return new KeyValuePair<string, object>(prop.Key.ToString(), prop.Value);
+                            }
+                        }
+                        else
+                        {
+                            index -= LogEvent.Properties.Count;
+                        }
+                    }
+                    if (ContextProperties != null)
+                    {
+                        if (index < ContextProperties.Count)
+                        {
+                            foreach (var prop in ContextProperties)
+                            {
+                                if (index-- == 0)
+                                    return new KeyValuePair<string, object>(prop.Key, prop.Value);
+                            }
+                        }
+                        else
+                        {
+                            index -= ContextProperties.Count;
+                        }
+                    }
+                    if (index != 0)
+                        throw new ArgumentOutOfRangeException(nameof(index));
+                    return CreateOriginalFormatProperty();
+                }
+            }
+
+            private KeyValuePair<string, object> CreateOriginalFormatProperty()
+            {
+                return new KeyValuePair<string, object>(NLogLogger.OriginalFormatPropertyName, LogEvent.Message);
+            }
+
+            public readonly LogEventInfo LogEvent;
+            public readonly string LayoutMessage;
+            public readonly IDictionary<string, object> ContextProperties;
```

Comment:
the fields are a bit of a surprise here. I really prefer: http://stylecop.soyuz5.com/SA1201.html

> Within a class, struct, or interface, elements must be positioned in the following order:

> Fields
Constructors
Finalizers (Destructors)
Delegates
Events
Enums
Interfaces
Properties
Indexers
Methods
Structs
Classes

Reply:
Moved code around

Reply:
Thanks!